### DISCUSSION TITLE
classes/class_os

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#238] [User-contributed notes]
# classes/class_os

Inherits: Object Provides access to common operating system functionalities. Description: The OS class wraps the most common functionalities for communicating with the host operating system, such a...

https://docs.godotengine.org/en/4.3/classes/class_os.html

**Comment by emanuelbertey** [comment]
command line does not work on Windows when I want to open another project in godot, it always goes to the godot.exe and runs the project and render mode, the cmd=true option does not work either, cmd does not open, do not document command line in the engine if it does not work, or choose a legend of this may not work

**Comment by JeanKouss** [comment]
When using `execute_with_pipe`, it is better to read the `stdout` in a dedicated thread.

When you try to read `stdout`, if data is available, it’s retrieved immediately. However, if `stdout` is empty, the reading thread will pause until data arrives. Attempting this in the main thread risks freezing your application if the external process takes time to produce output.
